### PR TITLE
Set session properties on session span on init

### PR DESCRIPTION
**Pull request by bidetofevil**
## Goal

Populate the session span with the current set session properties.

Add integration test to verify that this works end to end.

### CODE REVIEW

**Comment by bidetofevil** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/695?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#695** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/695?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#694** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/694?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#693** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/693?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#692** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/692?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#691** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/691?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#690** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/690?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @bidetofevil and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/session/SessionPayloadSourceImpl.kt [hunk 1]
```diff
@@ -21,21 +23,32 @@ internal class SessionPayloadSourceImpl(
...
+    private val sessionPropertiesServiceProvider: Provider<SessionPropertiesService>
 ) : SessionPayloadSource {
 
     override fun getSessionPayload(endType: SessionSnapshotType): SessionPayload {
+        val sharedLibSymbolMapping = captureDataSafely(logger) { nativeThreadSamplerService?.getNativeSymbols() }
+        val internalErrors = captureDataSafely(logger) { internalErrorService.currentExceptionError?.toNewPayload() }
+        val snapshots = retrieveSpanSnapshotData()
+
+        // Ensure the span retrieving is last as that potentially ends the session span, which effectively ends the session
+        val spans = retrieveSpanData(endType)
```
Comment (bidetofevil) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/session/SessionPayloadSourceImpl.kt:36]: This fixes a race condition that could cause a span snapshot of the NEXT session to be included in the payload for this session. Stopping the session span last ensures that doesn't happen.

File: embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/features/SessionPropertiesTest.kt [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import io.embrace.android.embracesdk.arch.schema.EmbType
+import io.embrace.android.embracesdk.internal.spans.hasFixedAttribute
+import io.embrace.android.embracesdk.recordSession
+import org.junit.Assert.assertEquals
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class SessionPropertiesTest {
```
Comment (fractalwrench) [inline embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/features/SessionPropertiesTest.kt:14]: Would suggest also adding test cases for `removeSessionProperty` and `getSessionProperties` for completeness

**Review by fractalwrench — APPROVED** [review]
LGTM

File: embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/features/SessionPropertiesTest.kt (hunk 2 above)
Comment (bidetofevil) [reply embrace-android-sdk/src/integrationTest/kotlin/io/embrace/android/embracesdk/features/SessionPropertiesTest.kt:14]: Ah yes, will do

**Pull request merged**